docs/cdp-options.md
---
id: chrome-options
title: Chrome
---

## Options

### --presetEnv

| Description                                                           | Type    | Default |
| :-------------------------------------------------------------------- | :------ | :------ |
| Preset the test environment with Sinon, Chai, Sinon-Chai, Chai subset | boolean | true    |

---

### --config

| Description         | Type   | Default | Alias |
| :------------------ | :----- | :------ | ----- |
| Path to config file | string |         | c     |

---

### --url

| Description      | Type   | Default |
| :--------------- | :----- | :------ |
| Url to html file | string |         |

---

### --scope

| Description      | Type  | Default | Alias |
| :--------------- | :---- | :------ | ----- |
| Scope to package | array |         | s     |

---

### --test

| Description  | Type  | Default                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       | Alias |
| :----------- | :---- | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----- |
| Glob pattern | array | commands/aw/**/*.{spec,test}.{js,ts},commands/cdp/**/*.{spec,test}.{js,ts},commands/cli/**/*.{spec,test}.{js,ts},commands/node/**/*.{spec,test}.{js,ts},commands/node-cli/**/*.{spec,test}.{js,ts},commands/protractor/**/*.{spec,test}.{js,ts},commands/puppeteer/**/*.{spec,test}.{js,ts},commands/serve/**/*.{spec,test}.{js,ts},examples/chrome-js/**/*.{spec,test}.{js,ts},examples/chrome-ts/**/*.{spec,test}.{js,ts},examples/node-js/**/*.{spec,test}.{js,ts},examples/node-js-no-babel/**/*.{spec,test}.{js,ts},examples/node-ts/**/*.{spec,test}.{js,ts},examples/protractor/**/*.{spec,test}.{js,ts},examples/puppeteer/**/*.{spec,test}.{js,ts},examples/react/**/*.{spec,test}.{js,ts},examples/webpack-dev-server/**/*.{spec,test}.{js,ts},packages/register/**/*.{spec,test}.{js,ts},packages/server/**/*.{spec,test}.{js,ts},packages/transform/**/*.{spec,test}.{js,ts},packages/transform-middleware/**/*.{spec,test}.{js,ts},packages/utils/**/*.{spec,test}.{js,ts},plugins/chai-plugin-screenshot/**/*.{spec,test}.{js,ts},plugins/chai-plugin-snapshot/**/*.{spec,test}.{js,ts},plugins/interactive-plugin/**/*.{spec,test}.{js,ts},plugins/preset-plugin/**/*.{spec,test}.{js,ts},plugins/watch-plugin/**/*.{spec,test}.{js,ts},!**/node_modules/**,!./node_modules/** | glob  |

---

### --testExt

| Description                       | Type   | Default               |
| :-------------------------------- | :----- | :-------------------- |
| Test file extensions glob pattern | string | *.{spec,test}.{js,ts} |

---

### --src

| Description                       | Type  | Default                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |
| :-------------------------------- | :---- | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| Glob pattern for all source files | array | commands/aw/**/*.{js,ts},commands/cdp/**/*.{js,ts},commands/cli/**/*.{js,ts},commands/node/**/*.{js,ts},commands/node-cli/**/*.{js,ts},commands/protractor/**/*.{js,ts},commands/puppeteer/**/*.{js,ts},commands/serve/**/*.{js,ts},examples/chrome-js/**/*.{js,ts},examples/chrome-ts/**/*.{js,ts},examples/node-js/**/*.{js,ts},examples/node-js-no-babel/**/*.{js,ts},examples/node-ts/**/*.{js,ts},examples/protractor/**/*.{js,ts},examples/puppeteer/**/*.{js,ts},examples/react/**/*.{js,ts},examples/webpack-dev-server/**/*.{js,ts},packages/register/**/*.{js,ts},packages/server/**/*.{js,ts},packages/transform/**/*.{js,ts},packages/transform-middleware/**/*.{js,ts},packages/utils/**/*.{js,ts},plugins/chai-plugin-screenshot/**/*.{js,ts},plugins/chai-plugin-snapshot/**/*.{js,ts},plugins/interactive-plugin/**/*.{js,ts},plugins/preset-plugin/**/*.{js,ts},plugins/watch-plugin/**/*.{js,ts},!**/node_modules/**,!./node_modules/** |

---

### --srcExt

| Description                       | Type   | Default |
| :-------------------------------- | :----- | :------ |
| Test file extensions glob pattern | string |         |

---

### --watch

| Description   | Type    | Default | Alias |
| :------------ | :------ | :------ | ----- |
| Watch changes | boolean | false   | w     |

---

### --watchGlob

| Description | Type  | Default                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 | Alias |
| :---------- | :---- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ----- |
| Watch glob  | array | commands/aw/**/*.{spec,test}.{js,ts},commands/cdp/**/*.{spec,test}.{js,ts},commands/cli/**/*.{spec,test}.{js,ts},commands/node/**/*.{spec,test}.{js,ts},commands/node-cli/**/*.{spec,test}.{js,ts},commands/protractor/**/*.{spec,test}.{js,ts},commands/puppeteer/**/*.{spec,test}.{js,ts},commands/serve/**/*.{spec,test}.{js,ts},examples/chrome-js/**/*.{spec,test}.{js,ts},examples/chrome-ts/**/*.{spec,test}.{js,ts},examples/node-js/**/*.{spec,test}.{js,ts},examples/node-js-no-babel/**/*.{spec,test}.{js,ts},examples/node-ts/**/*.{spec,test}.{js,ts},examples/protractor/**/*.{spec,test}.{js,ts},examples/puppeteer/**/*.{spec,test}.{js,ts},examples/react/**/*.{spec,test}.{js,ts},examples/webpack-dev-server/**/*.{spec,test}.{js,ts},packages/register/**/*.{spec,test}.{js,ts},packages/server/**/*.{spec,test}.{js,ts},packages/transform/**/*.{spec,test}.{js,ts},packages/transform-middleware/**/*.{spec,test}.{js,ts},packages/utils/**/*.{spec,test}.{js,ts},plugins/chai-plugin-screenshot/**/*.{spec,test}.{js,ts},plugins/chai-plugin-snapshot/**/*.{spec,test}.{js,ts},plugins/interactive-plugin/**/*.{spec,test}.{js,ts},plugins/preset-plugin/**/*.{spec,test}.{js,ts},plugins/watch-plugin/**/*.{spec,test}.{js,ts},!**/node_modules/**,!./node_modules/**,commands/aw/**/*.{js,ts},commands/cdp/**/*.{js,ts},commands/cli/**/*.{js,ts},commands/node/**/*.{js,ts},commands/node-cli/**/*.{js,ts},commands/protractor/**/*.{js,ts},commands/puppeteer/**/*.{js,ts},commands/serve/**/*.{js,ts},examples/chrome-js/**/*.{js,ts},examples/chrome-ts/**/*.{js,ts},examples/node-js/**/*.{js,ts},examples/node-js-no-babel/**/*.{js,ts},examples/node-ts/**/*.{js,ts},examples/protractor/**/*.{js,ts},examples/puppeteer/**/*.{js,ts},examples/react/**/*.{js,ts},examples/webpack-dev-server/**/*.{js,ts},packages/register/**/*.{js,ts},packages/server/**/*.{js,ts},packages/transform/**/*.{js,ts},packages/transform-middleware/**/*.{js,ts},packages/utils/**/*.{js,ts},plugins/chai-plugin-screenshot/**/*.{js,ts},plugins/chai-plugin-snapshot/**/*.{js,ts},plugins/interactive-plugin/**/*.{js,ts},plugins/preset-plugin/**/*.{js,ts},plugins/watch-plugin/**/*.{js,ts},!**/node_modules/**,!./node_modules/** | wg    |

---

### --coverage

| Description        | Type    | Default |
| :----------------- | :------ | :------ |
| Generate coverage? | boolean | false   |

---

### --hookRequire

| Description                                         | Type    | Default |
| :-------------------------------------------------- | :------ | :------ |
| Hook require to be able to mock and transform files | boolean | true    |

---

### --babel.enable

| Description  | Type    | Default |
| :----------- | :------ | :------ |
| Enable babel | boolean | true    |

---

### --babel.core

| Description               | Type   | Default |
| :------------------------ | :----- | :------ |
| Path to babel core module | string |         |

---

### --babel.babelPluginIstanbul

| Description                          | Type   | Default               |
| :----------------------------------- | :----- | :-------------------- |
| Path to babel plugin istanbul module | string | babel-plugin-istanbul |

---

### --babel.options

| Description   | Type   | Default         |
| :------------ | :----- | :-------------- |
| Babel options | object | [object Object] |

---

### --babel.typescript

| Description                        | Type   | Default    |
| :--------------------------------- | :----- | :--------- |
| Path to typescript compiler module | string | typescript |

---

### --transform.typescript.config

| Description            | Type   | Default       |
| :--------------------- | :----- | :------------ |
| Typescript config file | string | tsconfig.json |

---

### --transform.typescript.babelOptions

| Description              | Type   | Default         |
| :----------------------- | :----- | :-------------- |
| Typescript babel options | object | [object Object] |

---

### --transform.include

| Description                 | Type  | Default |
| :-------------------------- | :---- | :------ |
| Transform files with babel? | array |         |

---

### --transform.exclude

| Description  | Type  | Default |
| :----------- | :---- | :------ |
| Exclude glob | array |         |

---

### --transform.defaultExclude

| Description          | Type  | Default                                                                                                                                                                     |
| :------------------- | :---- | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| Default exclude glob | array | **/node_modules/**,./node_modules/**,**/coverage/**,**/external/**,**/autogenerated/**,**/*.{html,css,json,txt,ttf,woff,svg,ico},**/*require*.js,**/*sinon*.js,**/*chai*.js |

---

### --mocha.bail

| Description      | Type    | Default |
| :--------------- | :------ | :------ |
| Bails on failure | boolean | true    |

---

### --mocha.ui

| Description    | Type   | Default |
| :------------- | :----- | :------ |
| Test interface | string | bdd     |

---

### --mocha.timeout

| Description | Type   | Default |
| :---------- | :----- | ------: |
| Timeout     | number |    2000 |

---

### --mocha.color

| Description          | Type    | Default |
| :------------------- | :------ | :------ |
| Use colors in output | boolean | true    |

---

### --mocha.reporter

| Description | Type   | Default |
| :---------- | :----- | :------ |
| Reporter    | string | min     |

---

### --client.port

| Description | Type   | Default |
| :---------- | :----- | ------: |
| Chrome port | number |    9222 |

---

### --chrome.launch

| Description   | Type    | Default |
| :------------ | :------ | :------ |
| Launch Chrome | boolean | true    |

---

### --chrome.devtools

| Description                | Type    | Default |
| :------------------------- | :------ | :------ |
| Open Chrome with dev tools | boolean | false   |

---

### --chrome.port

| Description | Type   | Default |
| :---------- | :----- | ------: |
| Chrome port | number |    9222 |

---

### --chrome.chromeFlags

| Description  | Type  | Default                                                 |
| :----------- | :---- | :------------------------------------------------------ |
| Chrome flags | array | --headless,--disable-gpu,--allow-file-access-from-files |

---

### --http.port

| Description              | Type   | Default |
| :----------------------- | :----- | ------: |
| Listen on this http port | number |    9676 |

---

### --http.root

| Description           | Type  | Default |
| :-------------------- | :---- | :------ |
| Root folders to serve | array | ./      |

---

### --http.rewrite

| Description    | Type   | Default         |
| :------------- | :----- | :-------------- |
| Rewrite url(s) | object | [object Object] |

---

### --nyc.tempDirectory

| Description                                     | Type   | Default                |
| :---------------------------------------------- | :----- | :--------------------- |
| Directory to output raw coverage information to | string | ./coverage/.nyc_output |

---

### --nyc.reporter

| Description                 | Type  | Default           |
| :-------------------------- | :---- | :---------------- |
| Coverage reporter(s) to use | array | lcov,text-summary |

---

### --nyc.reportDir

| Description                             | Type   | Default  |
| :-------------------------------------- | :----- | :------- |
| Directory to output coverage reports in | string | coverage |

---

### --nyc.exclude

| Description  | Type  | Default                                                                                                                                                                     |
| :----------- | :---- | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| Exclude glob | array | **/node_modules/**,./node_modules/**,**/coverage/**,**/external/**,**/autogenerated/**,**/*.{html,css,json,txt,ttf,woff,svg,ico},**/*require*.js,**/*sinon*.js,**/*chai*.js |

---

### --interactive

| Description             | Type    | Default | Alias |
| :---------------------- | :------ | :------ | ----- |
| Run in interactive mode | boolean | false   | i     |

---

### --filter.chrome.packages

| Description                       | Type  | Default |
| :-------------------------------- | :---- | :------ |
| Filter packages for Chrome runner | array | **      |

---

### --filter.chrome.files

| Description                    | Type  | Default |
| :----------------------------- | :---- | :------ |
| Filter files for Chrome runner | array | **      |

---


CHANGELOG.md
<a name=""></a>
# [](https://github.com/qlik-oss/after-work.js/compare/v5.0.0...v) (2018-09-04)



<a name="5.0.0"></a>
# [5.0.0](https://github.com/qlik-oss/after-work.js/compare/v5.0.0-beta.7...v5.0.0) (2018-09-04)


### Features

* **puppeteer:** opt out on chrome stable ([#343](https://github.com/qlik-oss/after-work.js/issues/343)) ([5fad3d6](https://github.com/qlik-oss/after-work.js/commit/5fad3d6))



<a name="5.0.0-beta.7"></a>
# [5.0.0-beta.7](https://github.com/qlik-oss/after-work.js/compare/v5.0.0-beta.6...v5.0.0-beta.7) (2018-08-30)



<a name="5.0.0-beta.6"></a>
# [5.0.0-beta.6](https://github.com/qlik-oss/after-work.js/compare/v5.0.0-beta.5...v5.0.0-beta.6) (2018-08-23)


### Features

* **node:** trigger all tests ([#320](https://github.com/qlik-oss/after-work.js/issues/320)) ([7e5673b](https://github.com/qlik-oss/after-work.js/commit/7e5673b))
* local mocking in node ([#324](https://github.com/qlik-oss/after-work.js/issues/324)) ([212fa65](https://github.com/qlik-oss/after-work.js/commit/212fa65))



<a name="5.0.0-beta.5"></a>
# [5.0.0-beta.5](https://github.com/qlik-oss/after-work.js/compare/v5.0.0-beta.4...v5.0.0-beta.5) (2018-08-15)


### Bug Fixes

* chai-plugin-screenshot dependency ([#318](https://github.com/qlik-oss/after-work.js/issues/318)) ([0288081](https://github.com/qlik-oss/after-work.js/commit/0288081))



<a name="5.0.0-beta.4"></a>
# [5.0.0-beta.4](https://github.com/qlik-oss/after-work.js/compare/v5.0.0-beta.3...v5.0.0-beta.4) (2018-08-15)



<a name="5.0.0-beta.3"></a>
# [5.0.0-beta.3](https://github.com/qlik-oss/after-work.js/compare/v5.0.0-beta.2...v5.0.0-beta.3) (2018-08-15)


### Bug Fixes

* running typescript in node ([#308](https://github.com/qlik-oss/after-work.js/issues/308)) ([7b0562a](https://github.com/qlik-oss/after-work.js/commit/7b0562a))


### Features

* chai screenshot plugin ([#306](https://github.com/qlik-oss/after-work.js/issues/306)) ([8b6cb15](https://github.com/qlik-oss/after-work.js/commit/8b6cb15))
* puppeteer with watch ([#315](https://github.com/qlik-oss/after-work.js/issues/315)) ([31a7a29](https://github.com/qlik-oss/after-work.js/commit/31a7a29))



<a name="5.0.0-beta.2"></a>
# [5.0.0-beta.2](https://github.com/qlik-oss/after-work.js/compare/v4.3.2...v5.0.0-beta.2) (2018-06-28)


### Bug Fixes

* **protractor:** ensure setup tranformation ([#273](https://github.com/qlik-oss/after-work.js/issues/273)) ([c0ea515](https://github.com/qlik-oss/after-work.js/commit/c0ea515))
* ensure correct babel options ([#276](https://github.com/qlik-oss/after-work.js/issues/276)) ([5e82138](https://github.com/qlik-oss/after-work.js/commit/5e82138))
* move typescript option ([#272](https://github.com/qlik-oss/after-work.js/issues/272)) ([30a9b64](https://github.com/qlik-oss/after-work.js/commit/30a9b64))



<a name="4.3.2"></a>
## [4.3.2](https://github.com/qlik-oss/after-work.js/compare/v4.3.1...v4.3.2) (2018-05-30)


### Features

* cache transformed files ([#236](https://github.com/qlik-oss/after-work.js/issues/236)) ([37dc1d5](https://github.com/qlik-oss/after-work.js/commit/37dc1d5))



<a name="4.3.1"></a>
## [4.3.1](https://github.com/qlik-oss/after-work.js/compare/v4.3.0...v4.3.1) (2018-05-28)


### Bug Fixes

* lazy require `server` ([#235](https://github.com/qlik-oss/after-work.js/issues/235)) ([83e4770](https://github.com/qlik-oss/after-work.js/commit/83e4770))



<a name="4.3.0"></a>
# [4.3.0](https://github.com/qlik-oss/after-work.js/compare/v4.2.1...v4.3.0) (2018-05-24)


### Bug Fixes

* **package:** update sinon to version 5.0.10 ([#232](https://github.com/qlik-oss/after-work.js/issues/232)) ([c7210ad](https://github.com/qlik-oss/after-work.js/commit/c7210ad)), closes [#209](https://github.com/qlik-oss/after-work.js/issues/209)


### Features

* enable transform in protractor ([#220](https://github.com/qlik-oss/after-work.js/issues/220)) ([33345d3](https://github.com/qlik-oss/after-work.js/commit/33345d3))



<a name="4.2.1"></a>
## [4.2.1](https://github.com/qlik-oss/after-work.js/compare/v4.2.0...v4.2.1) (2018-05-15)



<a name="4.2.0"></a>
# [4.2.0](https://github.com/qlik-oss/after-work.js/compare/v4.1.0...v4.2.0) (2018-05-03)



<a name="4.1.0"></a>
# [4.1.0](https://github.com/qlik-oss/after-work.js/compare/v4.0.1...v4.1.0) (2018-01-26)


### Bug Fixes

* **protractor:** extend config with argv ([#131](https://github.com/qlik-oss/after-work.js/issues/131)) ([dd0e610](https://github.com/qlik-oss/after-work.js/commit/dd0e610))



<a name="4.0.1"></a>
## [4.0.1](https://github.com/qlik-oss/after-work.js/compare/v4.0.1-0...v4.0.1) (2018-01-10)



<a name="4.0.1-0"></a>
## [4.0.1-0](https://github.com/qlik-oss/after-work.js/compare/v4.0.0...v4.0.1-0) (2018-01-05)



<a name="4.0.0"></a>
# [4.0.0](https://github.com/qlik-oss/after-work.js/compare/v3.1.0...v4.0.0) (2017-12-11)



<a name="3.1.0"></a>
# [3.1.0](https://github.com/qlik-oss/after-work.js/compare/v3.0.1...v3.1.0) (2017-10-02)



<a name="3.0.1"></a>
## [3.0.1](https://github.com/qlik-oss/after-work.js/compare/v3.0.0...v3.0.1) (2017-09-22)



<a name="3.0.0"></a>
# [3.0.0](https://github.com/qlik-oss/after-work.js/compare/v2.2.0...v3.0.0) (2017-09-22)



<a name="2.2.0"></a>
# [2.2.0](https://github.com/qlik-oss/after-work.js/compare/v1.0.0...v2.2.0) (2017-07-13)





commands/cdp/src/options.js
const {
  TEST_GLOB,
  SRC_GLOB,
  WATCH_GLOB,
  DEFAULT_SRC_EXT_PATTERN,
  DEFAULT_TEST_EXT_PATTERN,
  DEFAULT_INSTRUMENT_EXCLUDE_PATTERN,
  DEFAULT_TRANSFORM_EXCLUDE_PATTERN,
} = require("@after-work.js/utils");

module.exports = {
  presetEnv: {
    description:
      "Preset the test environment with Sinon, Chai, Sinon-Chai, Chai subset",
    default: true,
    type: "boolean",
  },
  config: {
    description: "Path to config file",
    type: "string",
    default: null,
    alias: "c",
  },
  url: {
    description: "Url to html file",
    type: "string",
    default: null,
  },
  scope: {
    description: "Scope to package",
    default: [],
    type: "array",
    alias: "s",
  },
  test: {
    description: "Glob pattern",
    default: TEST_GLOB,
    type: "array",
    alias: "glob",
  },
  testExt: {
    description: "Test file extensions glob pattern",
    default: DEFAULT_TEST_EXT_PATTERN,
    type: "string",
  },
  src: {
    description: "Glob pattern for all source files",
    default: SRC_GLOB,
    type: "array",
    hidden: true,
  },
  srcExt: {
    description: "Source file extensions glob pattern",
    default: DEFAULT_SRC_EXT_PATTERN,
    type: "string",
    hidden: true,
  },
  watch: {
    description: "Watch changes",
    default: false,
    type: "boolean",
    alias: "w",
  },
  watchGlob: {
    description: "Watch glob",
    default: WATCH_GLOB,
    type: "array",
    alias: "wg",
  },
  coverage: {
    description: "Generate coverage?",
    default: false,
    type: "boolean",
  },
  hookRequire: {
    description: "Hook require to be able to mock and transform files",
    default: true,
    type: "boolean",
  },
  "babel.enable": {
    description: "Enable babel",
    default: true,
    type: "boolean",
  },
  "babel.core": {
    description: "Path to babel core module",
    default: "",
    type: "string",
  },
  "babel.babelPluginIstanbul": {
    description: "Path to babel plugin istanbul module",
    default: "babel-plugin-istanbul",
    type: "string",
  },
  "babel.options": {
    description: "Babel options",
    default: {},
    type: "object",
  },
  "babel.typescript": {
    description: "Path to typescript compiler module",
    default: "typescript",
    type: "string",
  },
  "transform.typescript.config": {
    description: "Typescript config file",
    default: "tsconfig.json",
    type: "string",
  },
  "transform.typescript.babelOptions": {
    description: "Typescript babel options",
    default: {},
    type: "object",
  },
  "transform.include": {
    description: "Transform files with babel?",
    default: [],
    type: "array",
  },
  "transform.exclude": {
    description: "Exclude glob",
    default: [],
    type: "array",
  },
  "transform.defaultExclude": {
    description: "Default exclude glob",
    default: DEFAULT_TRANSFORM_EXCLUDE_PATTERN,
    type: "array",
  },
  "mocha.bail": {
    description: "Bails on failure",
    default: true,
    type: "boolean",
  },
  "mocha.ui": {
    description: "Test interface",
    default: "bdd",
    type: "string",
  },
  "mocha.timeout": {
    description: "Timeout",
    default: 2000,
    type: "number",
  },
  "mocha.color": {
    description: "Use colors in output",
    default: true,
    type: "boolean",
  },
  "mocha.reporter": {
    description: "Reporter",
    default: "min",
    type: "string",
  },
  "client.port": {
    description: "Chrome port",
    default: 9222,
    type: "number",
  },
  "chrome.launch": {
    description: "Launch Chrome",
    default: true,
    type: "boolean",
  },
  "chrome.devtools": {
    description: "Open Chrome with dev tools",
    default: false,
    type: "boolean",
  },
  "chrome.port": {
    description: "Chrome port",
    default: 9222,
    type: "number",
  },
  "chrome.chromeFlags": {
    description: "Chrome flags",
    default: ["--headless", "--disable-gpu", "--allow-file-access-from-files"],
    type: "array",
  },
  "http.port": {
    description: "Listen on this http port",
    default: 9676,
    type: "number",
  },
  "http.root": {
    description: "Root folders to serve",
    default: ["./"],
    type: "array",
  },
  "http.rewrite": {
    description: "Rewrite url(s)",
    default: {},
    type: "object",
  },
  "nyc.tempDirectory": {
    description: "Directory to output raw coverage information to",
    default: "./coverage/.nyc_output",
    type: "string",
  },
  "nyc.reporter": {
    description: "Coverage reporter(s) to use",
    default: ["lcov", "text-summary"],
    type: "array",
  },
  "nyc.reportDir": {
    description: "Directory to output coverage reports in",
    default: "coverage",
    type: "string",
  },
  "nyc.exclude": {
    description: "Exclude glob",
    default: DEFAULT_INSTRUMENT_EXCLUDE_PATTERN,
    type: "array",
  },
  interactive: {
    description: "Run in interactive mode",
    default: false,
    type: "boolean",
    alias: "i",
  },
  "filter.chrome.packages": {
    description: "Filter packages for Chrome runner",
    default: ["**"],
    type: "array",
  },
  "filter.chrome.files": {
    description: "Filter files for Chrome runner",
    default: ["**"],
    type: "array",
  },
};
